db/db_range_del_test.cc

```diff
@@ -1521,6 +1521,88 @@ TEST_F(DBRangeDelTest, RangeTombstoneWrittenToMinimalSsts) {
   ASSERT_EQ(1, num_range_deletions);
 }
 
+TEST_F(DBRangeDelTest, OverlappedTombstones) {
+   const int kNumPerFile = 4, kNumFiles = 2;
+   Options options = CurrentOptions();
+   options.disable_auto_compactions = true;
+   options.max_compaction_bytes = 9*1024;
+   options.db_log_dir = "/tmp";
+   DestroyAndReopen(options);
+   // snapshot prevents key from being deleted during flush
+   const Snapshot* snapshot = db_->GetSnapshot();
```

Comment:
This snapshot is created against an empty DB. I don't think it prevents any key from being deleted.